File: Part2/Section 8 - Regresión con Arbol de Desicion/Decision Tree Regression.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sun Jan  5 17:21:13 2020

@author: root
"""

#Regresión con Arboles de Decision

#------<PREPROCESADO DE DATOS>----------

#Como importar una libreria

import  numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importar dataset
df=pd.read_csv("Position_Salaries.csv")

x=df.iloc[:, 1:2].values
y=df.iloc[:,-1].values
y=y.reshape(-1,1)
"""
#Escalado de variables
from sklearn.preprocessing import StandardScaler as scaler
scala_x=scaler()
x_train = scala_x.fit_transform(x_train)
x_test = scala_x.transform(x_test)
"""
#------<AJUSTAR MODELOS DE REGRESION>----------

#Ajustar regresion con Aroboles de Decision
from sklearn.tree import DecisionTreeRegressor
regression=DecisionTreeRegressor(random_state=0)
regression.fit(x,y)
print(regression)

#predicción de nuestros modelos

y_pred = regression.predict([[6.5]])


#------<VISUALIZAR MODELOS DE REGRESION>----------

#visualizacion de modelo sin suavisado
plt.scatter(x,y,color="red")
plt.plot(x,regression.predict(x), color="blue")
plt.title("Modelo de Regresión Lineal")
plt.xlabel("Posición del Empleado")
plt.ylabel("Sueldo (en u$s)")
plt.show()

"""
#visualizacion del modelo con suavizado
x_grid=np.arange(min(x),max(x),0.1)
x_grid=x_grid.reshape(len(x_grid),1)
plt.scatter(x,y,color="red")
plt.plot(x_grid,regression.predict(x_poli), color="blue")
plt.title("Modelo de Regresión")
plt.xlabel("Posición del Empleado")
plt.ylabel("Sueldo (en u$s)")
plt.show()
"""

File: Part2/Section 8 - Regresión con Arbol de Desicion/Decision Tree Regression.R
# Regression con Arbol de Decision

#--->>Pre Procesado de Datos

#Importar Dataset
df=read.csv("Position_Salaries.csv")
#filtrado de datasets (filas, columnas)
df=df[,2:3]

#-----Crear modelo de regresion-------

#Ajustar Modelo de regresion con Arbol de desicion
#Crear variables de regresion
library(rpart)
regression = rpart(formula=Salary ~ .,
              data=df,
              control = rpart.control(minsplit = 1))



#---------Realizar Predicción____________

#prediccion de nuevos resultados con Arbol de desicion
y_pred=predict(regression, newdata = data.frame(Level=6.5))

#------Crear Visualización de los datos________

#visualizacion modelo de Desicion
#visualizacion modelo de sin suavizado
library(ggplot2)
ggplot()+ 
  geom_point(aes(x=df$Level, y=df$Salary),
             color="red")+
  geom_line(aes(x=df$Level , y=predict(regression, newdata=data.frame(Level =df$Level))),
            color="blue")+
  ggtitle("Predicción con Arbol de Decision del Sueldo")+
  xlab("Nivel del Empleado")+
  ylab("Sueldo (USD)")


#visualizacion modelo de con suavizado

x_grid=seq(min(df$Level),max(df$Level),0.1)
ggplot()+ 
  geom_point(aes(x=df$Level, y=df$Salary),
             color="red")+
  geom_line(aes(x=x_grid , y=predict(regression, 
                                     newdata=data.frame(Level=x_grid))),
                                      color="blue")+
  ggtitle("Predicción con Arbol de Decision del Sueldo")+
  xlab("Nivel del Empleado")+
  ylab("Sueldo (USD)")

